### ISSUE TITLE
proposal: `--deep` / `--depth=XX` option.

### ISSUE THREAD

**Issue by jamestalmage** [tjunnone/npm-check-updates#154]
This would traverse the entire dependency tree and find ALL outdated dependencies throughout the entire graph. This would warn you if you are relying on a library that is not actively maintained and has not updated it's own dependencies in a long time. Hopefully this would drive some users to get involved filing PR's against their dependencies even if they are just simple `bump XXX` PR's.

There is is some value in deeply traversing the production dependencies of your dev dependencies, but maybe not dev dependencies of your dev dependencies.

**Comment by metaraine** [comment]
Thanks! That would be a nice addition. It would not be hard to call ncu recursively on nested package.json's. It may require some UI thoughtfulness to have an understandable output.

I am putting this at low priority unless others chime in and say this is a strongly desired feature.

**Comment by jamestalmage** [comment]
> It may require some UI thoughtfulness to have an understandable output

Perhaps it should mix the current output style and that of `npm ls`

```
├─┬ chai@^1.9.0 -> 1.9.2
│ ├── assertion-error@^1.0.0 -> 
│ └─┬ deep-eql@~0.1.3 -> ~0.3.0
│   └── type-detect@0.1.x -> 0.2.x
```

Use `chalk` to highlight lines that are behind.

Options should include listing everything (regardless of whether it is behind or not), or filtering just the lines that have a outdated dependency. In the filtering case, you would probably want to keep parents of outdated deps in the tree (but give them some light-gray "disabled" color that tells users to ignore them).

Probably the first step is to create a JSON output that captures all the details, then worry about how to format that.

**Comment by jamestalmage** [comment]
OK,
So `npm ls -json` already exports a deep dependency graph in json format.
From there, It is simply a matter of walking that JSON and finding latest for every dependency.

**Comment by jamestalmage** [comment]
`bower list -j` seems equally helpful

**Comment by raineorshine** [comment]
Closing for cleanup purposes and adding the [revive-me](https://github.com/tjunnone/npm-check-updates/issues?utf8=%E2%9C%93&q=label%3Arevive-me) label. Feel free to re-open if you are interested in working on a pull request.

**Comment by kachkaev** [comment]
This is now resolved, thanks to #785 🎉

**Comment by milahu** [comment]
not working with pnpm

-> glob should list the hidden node_modules/.pnpm/
